src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/AuthModal.java

```diff
@@ -26,36 +26,29 @@
   private WebElement forgottenPasswordLink;
 
   private WebDriver webDriver;
+  private String mainWindowHandle;
 
   public AuthModal(WikiaWebDriver webDriver){
     super();
     this.webDriver = webDriver;
-
+    waitForNewWindow();
+    this.mainWindowHandle =  this.webDriver.getWindowHandle();
+    for(String winHandle : this.webDriver.getWindowHandles()){
+      this.webDriver.switchTo().window(winHandle);
+    }
     PageFactory.initElements(webDriver, this);
   }
 
-  private void switchToFrame(){
-    webDriver.switchTo().frame(iFrame);
-  }
-
-  private void switchBack(){
-    webDriver.switchTo().defaultContent();
-  }
-
   public boolean isOpened(){
-    switchToFrame();
     boolean isOpenedResult = authModal.isDisplayed();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "isOpenedResult". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1488)